src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,107 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
```

Comment:
this method should include:
resize if needed
getNodeIndex
do some logic to link node

Don't need to create a lot of separate methods, or methods that depends on other and so on. It's too complicated and not readable